### PR TITLE
Feat/webclient and cosys json

### PR DESCRIPTION
**Pull request by dominikhorn93**
**Description**
Change CoSys and S3 to webClient and switch CoSys to JSON payload.

**Reference**
- https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/issues/it-at-m/digiwf-core/41
- https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/issues/it-at-m/digiwf-project/391
- https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/issues/it-at-m/digiwf-core/41

**Checklist**

- [ ] JUnit tests created / adjusted
- [ ] Changelog is adjusted
- [ ] [Documentations] are completed

### CODE REVIEW

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-example/src/main/resources/application-local.properties [hunk 1]
```diff
@@ -5,9 +5,9 @@ io.muenchendigital.digiwf.cosys.engineTopic=dwf-digiwf-engine-local-01
 io.muenchendigital.digiwf.cosys.topic=dwf-cosys-local-01
 digiwf.streaming.group="dwf-cosys-service"
 # cosys
-io.muenchendigital.digiwf.cosys.url=localhost:800
-io.muenchendigital.digiwf.cosys.merge.datafile=/root/multi
+io.muenchendigital.digiwf.cosys.url=http://cosysk007.srv.muenchen.de/cosys-eai/rest
```
Comment (martind260) [inline digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-example/src/main/resources/application-local.properties]: This is internal

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,25 +4,35 @@
 
 ### Added
 
-- (alw-integration): alw integration modules added
```
Comment (martind260) [inline CHANGELOG.md]: Changelog entry missing

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/api/MessageProcessor.java [hunk 3]
```diff
@@ -1,7 +1,7 @@
-package io.muenchendigital.digiwf.integration.cosys.api;
```
Comment (martind260) [inline digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/api/MessageProcessor.java:1]: The renaming of the root package TOGETHER with a rather big change makes it unnecessarily difficult for a reviewer to follow the relevant changes. I beg you to leave out such refactorings next time and postpone it to an extra PR. THX

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+     * @param generateDocument Data for generating documents
+     * @return
+     */
+    public Mono<byte[]> generateCosysDocument(final GenerateDocument generateDocument) {
+        try {
+            return this.generationApi.generatePdf(
+                    generateDocument.getGuid(),
+                    generateDocument.getClient(),
+                    generateDocument.getRole(),
+                    this.createFile("data", new Gson().toJson(generateDocument.getVariables()).getBytes(StandardCharsets.UTF_8)),
```
Comment (martind260) [inline digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java]: I liked it better when they were constants

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    false,
+                    this.createFile("merge", this.configuration.getMergeOptions()),
```
Comment (martind260) [inline digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java]: I liked it better when they were constants

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                    null,
+                    null
+            );
+        } catch (final Exception ex) {
+            log.error("Document could not be created.", ex);
+            throw new RuntimeException("Document could not be created.");
+        }
+    }
+
+    private File createFile(final String name, final byte[] content) throws IOException {
```
Comment (martind260) [inline digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java]: Isnt this a helper methods too?

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/integration/cosys/domain/service/CosysService.java [hunk 7]
```diff
@@ -1,144 +0,0 @@
...
-                    throw new RuntimeException("Document could not be saved.");
-                }
-            }
-        } catch (final DocumentStorageClientErrorException | DocumentStorageServerErrorException | DocumentStorageException ex) {
-            log.error("Document could not be saved.", ex);
-            throw new RuntimeException("Document could not be saved.");
-        }
-    }
-
-    private String createCosysUrl(final GenerateDocumentRequest request) {
```
Comment (martind260) [inline digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/integration/cosys/domain/service/CosysService.java:100]: All that stuff is gone, great!

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-example/src/main/java/io/muenchendigital/digiwf/cosys/integration/example/DigiwfCosysExampleApplication.java [hunk 8]
```diff
@@ -1,9 +1,9 @@
-package io.muenchendigital.digiwf.integration.cosys.example;
+package io.muenchendigital.digiwf.cosys.integration.example;
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
-@SpringBootApplication
+@SpringBootApplication()
```
Comment (martind260) [inline digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-example/src/main/java/io/muenchendigital/digiwf/cosys/integration/example/DigiwfCosysExampleApplication.java:6]: unnecessary

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-example/src/main/resources/application-local.properties [hunk 9]
```diff
@@ -5,9 +5,9 @@ io.muenchendigital.digiwf.cosys.engineTopic=dwf-digiwf-engine-local-01
...
+io.muenchendigital.digiwf.cosys.merge.datafile=//multi
 io.muenchendigital.digiwf.cosys.merge.inputLanguage=Deutsch
 io.muenchendigital.digiwf.cosys.merge.outputLanguage=Deutsch
 io.muenchendigital.digiwf.cosys.merge.keepFields=unresolved-ref
-io.muenchendigital.digiwf.cosys.ssoTokenClientSecret="test"
+# security
```
Comment (martind260) [inline digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-example/src/main/resources/application-local.properties:13]: pls remove

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-starter/src/main/java/io/muenchendigital/digiwf/cosys/integration/configuration/CosysAutoConfiguration.java [hunk 10]
```diff
@@ -60,5 +64,39 @@ public MessageRoutingCallback getEventBusRouter() {
...
+                        clientRegistrationRepository, authorizedClientService
+                )
+        );
+        oauth.setDefaultClientRegistrationId("cosys");
+        return WebClient.builder()
+                .baseUrl(this.cosysProperties.getUrl())
+                .exchangeStrategies(ExchangeStrategies.builder()
+                        .codecs(configurer -> configurer
+                                .defaultCodecs()
+                                .maxInMemorySize(32 * 1024 * 1024))
```
Comment (martind260) [inline digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-starter/src/main/java/io/muenchendigital/digiwf/cosys/integration/configuration/CosysAutoConfiguration.java:95]: Better putting this in the properties?

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-starter/pom.xml [hunk 11]
```diff
@@ -14,40 +14,50 @@
...
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter</artifactId>
-            <version>${spring.boot.version}</version>
         </dependency>
 
-        <!-- S3 -->
+        <dependency>
+            <groupId>org.springdoc</groupId>
+            <artifactId>springdoc-openapi-webflux-ui</artifactId>
+            <version>1.6.12</version>
```
Comment (martind260) [inline digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-starter/pom.xml]: Better put this in a property.

File: digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client/pom.xml [hunk 12]
```diff
@@ -42,8 +42,23 @@
         <!-- openApi -->
         <dependency>
             <groupId>org.springdoc</groupId>
-            <artifactId>springdoc-openapi-ui</artifactId>
-            <version>${springdoc.openapi.version}</version>
+            <artifactId>springdoc-openapi-webflux-ui</artifactId>
+            <version>1.6.12</version>
```
Comment (martind260) [inline digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client/pom.xml]: Better put all versions in properties

File: digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client-example/src/main/resources/logback-spring.xml [hunk 13]
```diff
@@ -1,119 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (martind260) [inline digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client-example/src/main/resources/logback-spring.xml:1]: Why did you remove this?

File: digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client-starter/src/main/java/io/muenchendigital/digiwf/s3/integration/client/configuration/S3IntegrationClientAutoConfiguration.java [hunk 14]
```diff
@@ -38,22 +43,38 @@ public class S3IntegrationClientAutoConfiguration {
...
-     * This factory class is providing either the preconfigured {@link FileApiApi} or {@link FileApiApi}.
-     *
-     * @param restTemplate to create rest requests.
-     *                     If the S3 integration service is secured via Oauth2,
-     *                     the OAuth2RestTemplate can be used here, for example.
-     * @return the {@link ApiClientFactory}.
-     */
     @Bean
-    public ApiClientFactory apiClientFactory(final RestTemplate restTemplate) {
+    @ConditionalOnProperty(prefix = "io.muenchendigital.digiwf.s3.client", name = "securityEnabled", havingValue = "true")
```
Comment (martind260) [inline digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client-starter/src/main/java/io/muenchendigital/digiwf/s3/integration/client/configuration/S3IntegrationClientAutoConfiguration.java:47]: You should mention this property in at least one properties-file to prevent having it undocumented.

**Review by martind260 — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (dominikhorn93) [reply CHANGELOG.md]: The Changelog is divided into apps (frontend) and services (backend). You can find the entry in the services part.

File: digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client-example/src/main/resources/logback-spring.xml (hunk 13 above)
Comment (dominikhorn93) [reply digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client-example/src/main/resources/logback-spring.xml:1]: added it again

File: digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client-starter/src/main/java/io/muenchendigital/digiwf/s3/integration/client/configuration/S3IntegrationClientAutoConfiguration.java (hunk 14 above)
Comment (dominikhorn93) [reply digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client-starter/src/main/java/io/muenchendigital/digiwf/s3/integration/client/configuration/S3IntegrationClientAutoConfiguration.java:47]: added it to the application-local.properties

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-starter/pom.xml (hunk 11 above)
Comment (dominikhorn93) [reply digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-starter/pom.xml]: done

File: digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client/pom.xml (hunk 12 above)
Comment (dominikhorn93) [reply digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-client/pom.xml]: done

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java (hunk 6 above)
Comment (dominikhorn93) [reply digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java]: done

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java (hunk 5 above)
Comment (dominikhorn93) [reply digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java]: done

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java (hunk 4 above)
Comment (dominikhorn93) [reply digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/domain/service/CosysService.java]: done

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/api/MessageProcessor.java (hunk 3 above)
Comment (dominikhorn93) [reply digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-core/src/main/java/io/muenchendigital/digiwf/cosys/integration/api/MessageProcessor.java:1]: sorry for that.

File: CHANGELOG.md (hunk 2 above)
Comment (ghost) [reply CHANGELOG.md]: OK, I see.

File: digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-example/src/main/resources/application-local.properties (hunk 1 above)
Comment (dominikhorn93) [reply digiwf-integrations/digiwf-cosys-integration/digiwf-cosys-integration-example/src/main/resources/application-local.properties]: done

**Review by ghost — APPROVED** [review]
